mitodl/mitxpro
# External/3rd Party Programs

#### Pre-Flight checklist


#### What are the relevant tickets?
fixes: #2034 

#### What's this PR do?
CMS users can now create external program pages. 

#### How should this be manually tested?
- Visit CMS
- Go into the program index page.
- Create a new `External Program Page` with relevant information there.
- Visit the `Catalog` page and see that the External Program are visible under `All/Programs` tab
- Verify the all relevant information is displaying properly as described in the ticket. #2034.
- Click any external program and see if the product page works fine with all the relevant details.

**Note:**
- To add a carousel view you can just visit the External Program Page in CMS and add it as a sub page - You'll need to check the `Override Content` checkbox if you want to add course pages in the carousel.
- Course pages list in the `Course Carousel Content` page in CMS now includes `External Course pages`, which can be added in Carousel & are displayed on Product Page.

#### Screenshots (if appropriate)
<img width="672" alt="Screen Shot 2021-01-11 at 6 47 01 PM" src="https://user-images.githubusercontent.com/7334669/104192098-2f1cf880-5440-11eb-97f1-8a4bec24168e.png">

![screencapture-localhost-8053-cms-pages-add-cms-externalprogrampage-37-2021-01-11-19_06_39](https://user-images.githubusercontent.com/7334669/104192118-36440680-5440-11eb-8aba-9bbb98bbe8ab.png)


<img width="897" alt="Screenshot 2021-01-11 at 6 57 14 PM" src="https://user-images.githubusercontent.com/7334669/104191305-28da4c80-543f-11eb-9f07-7a10cb14c4e0.png">

<img width="970" alt="Screenshot 2021-01-11 at 6 55 54 PM" src="https://user-images.githubusercontent.com/7334669/104191319-2e379700-543f-11eb-8a2a-6a28f87b155e.png">

<img width="1375" alt="Screenshot 2021-01-11 at 6 55 30 PM" src="https://user-images.githubusercontent.com/7334669/104191331-3263b480-543f-11eb-8078-5f86c6c2d4e0.png">

<img width="1332" alt="Screenshot 2021-01-13 at 4 26 14 AM" src="https://user-images.githubusercontent.com/34372316/104387189-654d9b80-5558-11eb-9972-24f5bf4620c2.png">

cc: @arslanashraf7
